### PR TITLE
base_fw: move SRAM bank details to Intel specific code

### PR DESCRIPTION
**Pull request by kv2019i**
The SRAM definitions like SRAM_BANK_SIZE and EBB_BANKS_IN_SEGMENT are only used in Intel specific code and not really needed in platform layer that needs to implemented by all platforms. Move these definitions to base_fw_intel.c and clean up related definitions from platform layer.

Relates to https://github.com/thesofproject/sof/issues/9015

### CODE REVIEW

File: src/platform/tigerlake/include/platform/lib/memory.h [hunk 1]
```diff
@@ -333,12 +333,6 @@
 #define SOF_STACK_SIZE		(CONFIG_SOF_STACK_SIZE)
 #define SOF_STACK_TOTAL_SIZE	(CONFIG_CORE_COUNT * SOF_STACK_SIZE)
 
-/* SOF Core S configuration */
-#define SOF_CORE_S_SIZE \
-	ALIGN((HEAP_SYSTEM_S_SIZE + HEAP_SYS_RUNTIME_S_SIZE + SOF_STACK_SIZE),\
-	SRAM_BANK_SIZE)
-#define SOF_CORE_S_T_SIZE ((CONFIG_CORE_COUNT - 1) * SOF_CORE_S_SIZE)
-
```
Comment (kv2019i) [inline src/platform/tigerlake/include/platform/lib/memory.h:341]: This is no longer needed and needs to be removed now when SRAM_BANK_SIZE is removed.

File: src/platform/posix/include/platform/lib/memory.h [hunk 2]
```diff
@@ -39,11 +39,6 @@ static inline void *platform_shared_get(void *ptr, int bytes)
 	return ptr;
 }
 
-#define SRAM_BANK_SIZE 0x10000
-#define EBB_BANKS_IN_SEGMENT 32
-#define PLATFORM_HPSRAM_EBB_COUNT 32
-#define PLATFORM_LPSRAM_EBB_COUNT 1
-
```
Comment (kv2019i) [inline src/platform/posix/include/platform/lib/memory.h:46]: This is no longer needed as these definitions are only used in the Intel specific basefw code (so not needed to build for posix).

File: src/platform/intel/cavs/include/cavs/lib/memory.h [hunk 3]
```diff
@@ -19,52 +19,6 @@
 /* data cache line alignment */
 #define PLATFORM_DCACHE_ALIGN	DCACHE_LINE_SIZE
 
-#define SRAM_BANK_SIZE			(64 * 1024)
```
Comment (kv2019i) [inline src/platform/intel/cavs/include/cavs/lib/memory.h:22]: This definition is in cavs25/adsp_memory.h in Zephyr, not needed here in SOF codebase. All below stuff is unused stuff that can be removed.

**Review by kv2019i — COMMENTED** [review]
Some inline comments to explain what is removed to help with the review.

**Review by lgirdwood — APPROVED** [review]

**Review by softwarecki — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX good to merge ? CI has been pending a while.

**Pull request merged**